### PR TITLE
feat(book list)

### PR DESCRIPTION
**Pull request by YgRamos9**
## Summary
- add render book list

## Screenshot

https://user-images.githubusercontent.com/107204529/184238425-a91939e4-fe2d-4dab-8a4c-ab31206ac529.mov

## Trello Card
https://trello.com/c/mVkYVzdy/6-book-list-auth-header

### CODE REVIEW

File: angular/src/app/components/book-card/book-card.component.scss [hunk 1]
```diff
@@ -0,0 +1,49 @@
+@import 'src/styles/variables/colors';
+@import 'src/styles/variables/fonts';
+
+@mixin text-ellipsis {
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+}
```
Comment (joacomiguens) [inline angular/src/app/components/book-card/book-card.component.scss]: Don't use mixin, add a new class

File: angular/src/styles/commons/buttons.scss [hunk 2]
```diff
@@ -45,3 +45,18 @@
     color: $earls-green;
   }
 }
+
+.cart-add {
+  background: $earls-green;
+  border-radius: 2em;
```
Comment (joacomiguens) [inline angular/src/styles/commons/buttons.scss]: use px

File: angular/src/app/pipes/filter-books.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { IBook } from '../interfaces/global.interface';
+
+@Pipe({
+  name: 'filterBooks',
+})
+export class FilterBooksPipe implements PipeTransform {
+  transform(value: IBook[], arg: string): IBook[] {
+    if (value) {
```
Comment (joacomiguens) [inline angular/src/app/pipes/filter-books.pipe.ts]: Delete this if statement, and add ? on value?.filter 😄

File: angular/src/app/screens/auth/auth-routing.module.ts [hunk 4]
```diff
@@ -1,13 +1,23 @@
...
 const routes: Routes = [
   {
     path: '',
     component: AuthComponent,
-    canActivate: [AuthGuard],
+    children: [
+      {
+        path: 'books',
+        component: BookListComponent,
+      },
```
Comment (joacomiguens) [inline angular/src/app/screens/auth/auth-routing.module.ts]: BookList should be a module, like login or sign-up

File: angular/src/app/screens/auth/screens/book-list/book-list.component.html [hunk 5]
```diff
@@ -1 +1,19 @@
-<p>book-list works!</p>
+<div class="book-search">
+  <input class="book-search-input" name="inputSearch" role="search" [(ngModel)]="inputSearch" type="search"
+    placeholder="Buscá por título del libro..." /><img class="book-icon-search" src="assets/search-icon.png"
```
Comment (joacomiguens) [inline angular/src/app/screens/auth/screens/book-list/book-list.component.html]: img tag should be on next line.

File: angular/src/app/screens/auth/screens/book-list/book-list.component.html [hunk 6]
```diff
@@ -1 +1,19 @@
...
+    alt="Search icon" />
+</div>
+
+<div class="books-container">
+  <ng-container *ngIf="(books | filterBooks: inputSearch).length === 0; else elseTemplate">
```
Comment (joacomiguens) [inline angular/src/app/screens/auth/screens/book-list/book-list.component.html]: ngIf logic should be on component file

Comment (joacomiguens) [reply angular/src/app/screens/auth/screens/book-list/book-list.component.html]: Also, elseTemplate should be the worse case scenario. So I think we could change condition 👍🏻

File: angular/src/app/screens/auth/screens/book-list/book-list.component.scss [hunk 7]
```diff
@@ -0,0 +1,45 @@
+@import 'src/styles/variables/colors';
+@import 'src/styles/variables/fonts';
+
+@mixin text-ellipsis {
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+}
```
Comment (joacomiguens) [inline angular/src/app/screens/auth/screens/book-list/book-list.component.scss]: same here and reuse that new class you are going to create 🔝

File: angular/src/app/screens/auth/screens/book-list/book-list.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+import { FormsModule } from '@angular/forms';
+import { render } from '@testing-library/angular';
+import { fireEvent, screen } from '@testing-library/dom';
+import '@testing-library/jest-dom';
+import { BookListComponent } from './book-list.component';
+import { BookCardComponent } from '../../../../components/book-card/book-card.component';
+import { mockBook } from '../../../../helpers/mocks/mock-books';
+import { FilterBooksPipe } from '../../../../pipes/filter-books.pipe';
+
+describe('BoolistComponent ', () => {
```
Comment (joacomiguens) [inline angular/src/app/screens/auth/screens/book-list/book-list.component.spec.ts]:
```suggestion
describe('BookListComponent ', () => {
```

File: angular/src/app/screens/un-auth/un-auth-routing.module.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  },
+  {
+    path: 'login',
+    loadChildren: () =>
+      import('./screens/login/login.module').then((m) => m.LoginModule),
+  },
+  {
+    path: '',
+    pathMatch: 'full',
+    redirectTo: 'login',
```
Comment (joacomiguens) [inline angular/src/app/screens/un-auth/un-auth-routing.module.ts]: use enum 🛩️

File: angular/src/app/services/books.service.ts [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+export class BooksService {
+  private API: string = environment.apiUrl;
+
+  constructor(private http: HttpClient) {}
+
+  getBooks(): Observable<HttpResponse<IBook[]>> {
+    return this.http.get(`${this.API}/api/v1/books`).pipe(
+      map((response: any) => {
+        return response.page;
+      }),
```
Comment (joacomiguens) [inline angular/src/app/services/books.service.ts]:
```suggestion
      map((response: any) => response.page),
```

File: angular/src/app/services/user.service.ts [hunk 11]
```diff
@@ -42,7 +42,19 @@ export class UserService {
...
+  getToken(): IAuthUser {
+    const activeToken = localStorage.getItem('userToken');
+    if (activeToken) {
+      return JSON.parse(activeToken);
+    }
+    return {
+      token: '',
+      client: '',
+      uid: '',
+    };
```
Comment (joacomiguens) [inline angular/src/app/services/user.service.ts]:
```suggestion
    return localStorage.getItem('userToken') || {
      token: '',
      client: '',
      uid: '',
    };
```

File: angular/src/app/interfaces/global.interface.ts [hunk 12]
```diff
@@ -20,3 +20,15 @@ export interface IAuthUser {
...
+  author: string;
+  created_at: string;
+  editor: string;
+  genre: string;
+  id: number;
+  image_url: string;
+  title: string;
+  updated_at: string;
+  year: string;
+}
```
Comment (joacomiguens) [inline angular/src/app/interfaces/global.interface.ts]: This interface should be on camelCase format. So serialize request response 💯

File: angular/src/app/components/book-card/book-card.component.html [hunk 13]
```diff
@@ -0,0 +1,6 @@
+<div class="books">
+  <img class="book-cover" [src]="book.image_url" [alt]="book.title" />
+  <div class="book-title">{{ book.title }}</div>
+  <div class="book-author">{{ book.author }}</div>
```
Comment (miguelhenaoa) [inline angular/src/app/components/book-card/book-card.component.html]: use span

File: angular/src/app/screens/auth/screens/book-list/book-list.component.html [hunk 14]
```diff
@@ -1 +1,19 @@
-<p>book-list works!</p>
+<div class="book-search">
+  <input class="book-search-input" name="inputSearch" role="search" [(ngModel)]="inputSearch" type="search"
```
Comment (miguelhenaoa) [inline angular/src/app/screens/auth/screens/book-list/book-list.component.html]: Format this tag, one property by line

File: angular/src/app/screens/un-auth/un-auth.component.html [hunk 15]
```diff
@@ -0,0 +1 @@
+<p>un-auth works!</p>
```
Comment (miguelhenaoa) [inline angular/src/app/screens/un-auth/un-auth.component.html]: is that what this is used for?

**Review by miguelhenaoa — CHANGES_REQUESTED** [review]

File: angular/src/app/screens/auth/screens/book-list/book-list.component.ts [hunk 16]
```diff
@@ -1,7 +1,25 @@
...
+
+  constructor(private booksService: BooksService) {}
+
+  ngOnInit(): void {
+    this.booksService.getBooks().subscribe((response: any) => {
+      this.books = response;
+    });
+  }
+
+  trackByBooks(index: number, item: IBook): number {
```
Comment (miguelhenaoa) [inline angular/src/app/screens/auth/screens/book-list/book-list.component.ts:22]: Remove unused param

File: angular/src/app/screens/auth/screens/book-list/book-list.component.ts [hunk 17]
```diff
@@ -1,7 +1,25 @@
...
 
 @Component({
   selector: 'app-book-list',
   templateUrl: './book-list.component.html',
+  styleUrls: ['./book-list.component.scss'],
 })
-export class BookListComponent {}
+export class BookListComponent implements OnInit {
+  books!: IBook[];
+  inputSearch!: any;
```
Comment (miguelhenaoa) [inline angular/src/app/screens/auth/screens/book-list/book-list.component.ts]: Add type

File: angular/src/app/screens/auth/screens/book-list/book-list.component.scss [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+
+.book-icon-search {
+  height: 24px;
+  max-width: 100%;
+  transform: translate(-40px, 13px);
+  width: 24px;
+}
+
+.book-empy {
```
Comment (miguelhenaoa) [inline angular/src/app/screens/auth/screens/book-list/book-list.component.scss]:
```suggestion
.book-empty {
```

File: angular/src/app/screens/auth/screens/book-list/book-list.component.scss [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+.book-search {
+  display: flex;
+  justify-content: center;
+  margin: 35px 0 45px;
+}
+
+.book-search-input {
+  border: 2px solid #e8e8e8;
```
Comment (miguelhenaoa) [inline angular/src/app/screens/auth/screens/book-list/book-list.component.scss]: this to colors.scss

File: angular/src/app/services/books.service.ts [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+import { map, Observable } from 'rxjs';
+import { environment } from '../../environments/environment';
+import { camelCaseSerializer } from '../helpers/utilities/serializer';
+import { IBook } from '../interfaces/global.interface';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class BooksService {
+  private API: string = environment.apiUrl;
```
Comment (miguelhenaoa) [inline angular/src/app/services/books.service.ts]:
```suggestion
  private readonly API: string = environment.apiUrl;
```

**Review by miguelhenaoa — CHANGES_REQUESTED** [review]

File: angular/src/app/helpers/mocks/mock-books.ts [hunk 21]
```diff
@@ -0,0 +1,37 @@
...
+    id: 3,
+    author: 'Sin registro',
+    title: 'The West Wing',
+    image_url: 'https://covers.openlibrary.org/b/id/476938-L.jpg',
+    editor: 'Pocket',
+    year: '8',
+    genre: 'Sin registro',
+    created_at: '2020-05-07T03:05:01.219Z',
+    updated_at: '2020-05-07T03:05:01.219Z',
+  },
```
Comment (joacomiguens) [inline angular/src/app/helpers/mocks/mock-books.ts]: should be in camelCase

File: angular/src/app/screens/auth/screens/book-list/book-list.component.ts [hunk 22]
```diff
@@ -1,7 +1,31 @@
...
+    });
+  }
+
+  trackByBooks(index: number, item: IBook): number {
+    return item.id;
+  }
+
+  get hasBooks(): boolean {
+    return this.pipe.transform(this.books, this.inputSearch).length > 0;
+  }
```
Comment (joacomiguens) [inline angular/src/app/screens/auth/screens/book-list/book-list.component.ts]: should be after ngOnInit

**Review by joacomiguens — APPROVED** [review]

**Review by miguelhenaoa — APPROVED** [review]
LGTM!! 🚀

**Pull request merged**